layers-samples/media-layer-sample.html

```diff
@@ -0,0 +1,204 @@
+
+      // XR globals.
+      let xrButton = document.getElementById('xr-button');
+      let xrSession = null;
+      let xrRefSpace = null;
+      let xrMediaFactory = null;
+      let cylinder_layer = false;
+
+      let video = document.createElement('video');
+      video.loop = false;
+      video.crossOrigin = "anonymous";
+      video.preload = 'auto';
+      video.autoload = true;
+      function buttonPressedThisFrame(gamepad, index) {
+        return (index < gamepad.buttons.length &&
+          gamepad.buttons[index].pressed);
+      }
+      function videoSelect() {
+        let s = document.getElementById("videoselect");
+        video.src = s.options[s.selectedIndex].value;
+        xrButton.textContent = 'Loading video';
+        xrButton.disabled = true;
+      }
+
+      video.oncanplaythrough = (event) => {
+        xrButton.disabled = false;
+        xrButton.textContent = 'Enter XR';
+      }
+
+      video.onerror = (event) => {
+        xrButton.disabled = true;
+        xrButton.textContent = 'Can\'t play video' + video.error.message;
+      };
+
+      videoSelect();
+      document.getElementById("videoselect").addEventListener('change', (event) => {
+        videoSelect();
+      });
+
+      // Checks to see if WebXR is available and, if so, requests an XRDevice
+      // that is connected to the system and tests it to ensure it supports the
+      // desired session options.
+      function initXR() {
+        // Is WebXR available on this UA?
+        if (navigator.xr) {
+          // If the device allows creation of exclusive sessions set it as the
+          // target of the 'Enter XR' button.
+          navigator.xr.isSessionSupported('immersive-vr').then((supported) => {
+            if (supported) {
+              // Updates the button to start an XR session when clicked.
+              xrButton.addEventListener('click', onButtonClicked);
+              xrButton.disabled = video.readyState > 2;
+            }
+          });
+        }
+      }
+
+      // Called when the user clicks the button to enter XR. If we don't have a
+      // session we'll request one, and if we do have a session we'll end it.
+      function onButtonClicked() {
+        if (!xrSession) {
+          navigator.xr.requestSession('immersive-vr', {requiredFeatures: ['layers']}).then(onSessionStarted);
+          video.play();
+        } else {
+          xrSession.end();
+        }
+      }
+
+      // Called when we've successfully acquired a XRSession. In response we
+      // will set up the necessary session state and kick off the frame loop.
+      function onSessionStarted(session) {
+        xrSession = session;
+        xrButton.textContent = 'Exit VR';
+
+        // Listen for the sessions 'end' event so we can respond if the user
+        // or UA ends the session for any reason.
+        session.addEventListener('end', onSessionEnded);
+
+        xrMediaFactory = new XRMediaBinding(session);
```

Comment:
Just a note to say that I really love how simple this interface is to use in practice! 👍